src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java

```diff
@@ -159,45 +160,102 @@ private <T> void injectBeanViaAutowiredAnnotation(Class<?> beanClass, T beanInst
-                var autowiredBeansInstance = getBean(parameterType);
-                String parameterName = fieldTypeStringNameMap.get(parameterType);
-                Field field = beanClass.getDeclaredField(parameterName);
-                field.setAccessible(true);
-                field.set(beanInstance, autowiredBeansInstance);
+            for (Class<?> parameterType : constructor.getParameterTypes()) {
+                assignBeans(beanClass, beanInstance, fieldTypeStringNameMap, parameterType);
+
             }
         }
     }
+
     /**
      * <h3>Injects Bean via Setter </h3>
      *
-     * @param beanClass    {@link Class}
-     * @param beanInstance {@link Object}
+     * @param beanClass              {@link Class}
+     * @param beanInstance           {@link Object}
+     * @param fieldTypeStringNameMap {@link Map}
      */
 
     private <T> void injectBeanViaSetter(Class<?> beanClass, T beanInstance, Map<Class<?>, String> fieldTypeStringNameMap) throws NoSuchFieldException, IllegalAccessException {
         Method[] beanMethods = beanClass.getDeclaredMethods();
-        for (Method method: beanMethods){
-            if (method.getName().startsWith("set") && method.isAnnotationPresent(Autowired.class)){
+        for (Method method : beanMethods) {
+            if (method.getName().startsWith("set") && method.isAnnotationPresent(Autowired.class)) {
                 Class<?>[] parameterTypes = method.getParameterTypes();
-                for (Class<?> parameterType: parameterTypes){
-                    var autowiredBeansInstance = getBean(parameterType);
-                    String parameterName = fieldTypeStringNameMap.get(parameterType);
-                    Field field = beanClass.getDeclaredField(parameterName);
-                    field.setAccessible(true);
-                    field.set(beanInstance, autowiredBeansInstance);
+                for (Class<?> parameterType : parameterTypes) {
+                    assignBeans(beanClass, beanInstance, fieldTypeStringNameMap, parameterType);
+
                 }
             }
         }
     }
+    /**
+     * <h3>Assign bean instance to the field of the class</h3>
+     * <p>Assigns the bean instance to the parent Class field </p>
+     * <p>{@link AnnotationConfigApplicationContext#multipleBeansExpected(Class)} is used to check whether {@link List} of beans are expected</p>
+     * <p>If list is expected then {@link AnnotationConfigApplicationContext#getElementListType(Class)} is used to determine parameter of the elements in {@link List}</p>
+     * <p>If one bean expected, {@link Field} is assigned with bean instance value</p>
+     *
+     * @param beanClass {@link Class}
+     * @param beanInstance {@link Object}
+     * @param fieldTypeStringNameMap {@link Map}
+     */
+    private <T> void assignBeans(Class<?> beanClass, T beanInstance, Map<Class<?>, String> fieldTypeStringNameMap, Class<?> parameterType) throws NoSuchFieldException, IllegalAccessException {
```

Comment:
Better to rename to injectBeans. What do you think?

Reply:
agree. will rename